lib/zlib.js

```diff
@@ -430,6 +430,11 @@ Zlib.prototype.close = function close(callback) {
   this.destroy();
 };
 
+Zlib.prototype._destroy = function _destroy(err, callback) {
+  _close(this);
+  callback(err);
```

Comment:
The `_destroy` docs on streams are a little fuzzy on this, but I've always assumed that you should only pass an error to the `callback` in case something went wrong during the destroy operation?

Reply:
This should be the right way to do it.